src/Symfony/Tests/HmacClient.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+namespace Acquia\Hmac\Symfony\Tests;
+
+use Acquia\Hmac\Key;
+use Acquia\Hmac\RequestSigner;
+use Acquia\Hmac\ResponseAuthenticator;
+use Symfony\Bundle\FrameworkBundle\Client;
+use Symfony\Bridge\PsrHttpMessage\Factory\DiactorosFactory;
+use Symfony\Bridge\PsrHttpMessage\Factory\HttpFoundationFactory;
+
+class HmacClient extends Client
+{
+
+    /**
+     * @var Key
+     */
+    protected $key;
+
+    /**
+     * Set Key for HMAC Auth
+     *
+     * @param Key $key
+     * @return Key
+     */
+    public function setKey(Key $key) {
+
+        $this->key = $key;
+
+        return $this;
+
+    }
+
+    /**
+     * Sign the request with HMAC HTTP and verify the response signature
+     *
+     * @param \Symfony\Component\HttpFoundation\Request $request
+     * @return \Symfony\Component\HttpFoundation\Response
+     * @throws \Exception
+     */
+    protected function doRequest($request)
+    {
+
+        if(!$this->key instanceof Key) {
+            throw new \Exception("Key not provided");
```

Comment:
Extending classes should not strengthen preconditions (it's a [Liskov Substitution Principle](https://en.wikipedia.org/wiki/Liskov_substitution_principle) violation). Instead this should just return a 401 response. For the response message, use single quotation marks and include a period.